sweep/tx_input_set.go

```diff
@@ -122,3 +171,71 @@ func (b *txInputSet) addPositiveYieldInputs(sweepableInputs []txInput,
 
 	// We managed to add all inputs to the set.
 }
+
+// tryAddWalletInputsIfNeeded retrieves utxos from the wallet and tries adding as
+// many as required to bring the tx output value above the given minimum.
+func (b *txInputSet) tryAddWalletInputsIfNeeded() error {
+	// If we've already reached the dust limit, no action is needed.
+	if b.dustLimitReached() {
+		return nil
+	}
+
+	// Retrieve wallet utxos. Only consider confirmed utxos to prevent
+	// problems around RBF rules for unconfirmed inputs.
+	utxos, err := b.wallet.ListUnspentWitness(1, math.MaxInt32)
+	if err != nil {
+		return err
+	}
+
+	for _, utxo := range utxos {
+		input, err := createWalletTxInput(utxo)
```

Comment:
Don't we want to lock these if they get chosen?

Reply:
If we are actually sweeping, this is executed within a coin selection lock. So no other processes should be selecting coins then.